src/FactorGraph01.jl

```diff
@@ -1017,19 +1015,21 @@ function addFactor!(dfg::G,
                       {G <: AbstractDFG,
                        R <: Union{FunctorInferenceType, InferenceType}}
   #
+  varOrderLabels = [v.label for v=Xi]
   namestring = assembleFactorName(dfg, Xi, maxparallel=maxparallel)
-  newFactor = DFGFactor{CommonConvWrapper{R}, Symbol}(Symbol(namestring))
-  newFactor.tags = union(labels, [:FACTOR]) # TODO: And session info
-  newFactor.solvable = solvable
-  # addNewFncVertInGraph!(fgl, newvert, currid, namestring, solvable)
-  newData = setDefaultFactorNode!(dfg, newFactor, Xi, deepcopy(usrfnc), multihypo=multihypo, threadmodel=threadmodel)
+  solverData = getDefaultFactorData(dfg, Xi, deepcopy(usrfnc), multihypo=multihypo, threadmodel=threadmodel)
+  newFactor = DFGFactor(Symbol(namestring),
+                        varOrderLabels,
+                        solverData;
+                        tags=Set(union(labels, [:FACTOR])),
+                        solvable=solvable)
 
   # TODO: Need to remove this...
   for vert in Xi
-    push!(newData.fncargvID, vert.label) # vert._internalId # YUCK :/ -- Yup, this is a problem
+    push!(solverData.fncargvID, vert.label)
```

Comment:
I tried to follow through on @GearsAD changes,. Please just confirm if this is what you wanted.

Reply:
defer this to @GearsAD please